client/my-sites/checkout/utils.ts

```diff
@@ -135,3 +144,49 @@ export function getAffiliateCouponLabel(): string {
 	// translators: The label of the coupon line item in checkout
 	return translate( 'Exclusive Offer Applied' );
 }
+
+export function useGetWpcomPlanTotalIfPaidMonthly( products: Array< ResponseCartProduct > ): {
+	[ key: string ]: number | undefined;
+} {
+	const plansToMonthly = products
+		.filter( ( product ) => {
+			return isWpComPlan( product?.product_slug ) && ! isMonthly( product?.product_slug );
+		} )
+		.map( ( product ) => {
+			return {
+				plan: product.product_slug,
+				monthly_plan: getMonthlyPlanByYearly( product.product_slug ),
+			};
+		} );
+	// Get unique non-empty monthly plan slugs
+	const monthlyPlanSlugs = [
+		...new Set( plansToMonthly.map( ( plan ) => plan.monthly_plan ).filter( ( plan ) => plan ) ),
+	] as PlanSlug[];
+	const pricing =
+		Plans.usePricingMetaForGridPlans( {
+			planSlugs: monthlyPlanSlugs,
+			siteId: undefined,
+			coupon: undefined,
+			useCheckPlanAvailabilityForPurchase,
+		} ) || {};
+	const plansToMonthlyPricing =
+		Object.fromEntries(
+			plansToMonthly.map( ( plan ) => {
+				return [ plan.plan, pricing[ plan.monthly_plan ] ?? null ];
+			} )
+		) || {};
```

Comment:
Do we need this intermediate structure too? You have everything in `pricing` already, no?

Reply:
Should be fixed now.